src/components/MultiHopTrade/components/TradeInput/components/TrendingTokens.tsx

```diff
@@ -0,0 +1,170 @@
+    (row: Row<Asset>) => {
+      const mixpanel = getMixPanel()
+
+      mixpanel?.track(MixPanelEvent.TrendingTokenClicked, {
+        assetId: row.original.assetId,
+        asset: row.original.symbol,
+        name: row.original.name,
+      })
+
+      dispatch(tradeInput.actions.setBuyAsset(row.original))
+    },
+    [dispatch],
+  )
+
+  useEffect(() => {
+    const updateMaxHeight = () => {
+      if (document.querySelector('.trade-amount-input')) {
+        const tradeAmountInputRect = document
+          .querySelector('.trade-amount-input')
+          ?.getBoundingClientRect()
+        const headerRect = document.querySelector('.swapper-header')?.getBoundingClientRect()
+        const swapperDividerRect = document
+          .querySelector('.swapper-divider')
+          ?.getBoundingClientRect()
+        const titleRect = titleRef.current?.getBoundingClientRect()
+        if (!tradeAmountInputRect || !headerRect || !swapperDividerRect || !titleRect) return
+
+        const bodyHeaderHeight =
+          tradeAmountInputRect.height * 2 +
+          headerRect.height +
+          swapperDividerRect.height +
+          titleRect.height
+        setBodyHeaderHeight(`${bodyHeaderHeight}px`)
+      }
+    }
+
+    // call once to set initial value
+    updateMaxHeight()
+
+    // update when window resizes
+    window.addEventListener('resize', updateMaxHeight)
+
+    // cleanup event listener
+    return () => window.removeEventListener('resize', updateMaxHeight)
+  }, [])
+
+  return (
+    <Box>
+      <Text
+        ref={titleRef}
+        color='text.primary'
+        fontWeight='bold'
+        translation='common.trendingTokens'
+        textDecoration='underline'
+        textUnderlineOffset='4px'
+        mb={4}
+        px={5}
+      />
+      {isTrendingAssetsLoading ? (
+        <Flex flexDir='column' width='100%' pb='var(--mobile-nav-offset)'>
+          {range(5).map(index => (
+            <Flex
+              key={index}
+              align='center'
+              width='100%'
+              justifyContent='space-between'
+              px={4}
+              mb={4}
+            >
+              <Flex align='center'>
+                <Skeleton width='40px' height='40px' borderRadius='100%' me={2} />
+                <Flex flexDir='column' gap={2}>
+                  <Skeleton width='140px' height='18px' />
+                  <Skeleton width='80px' height='18px' />
+                </Flex>
+              </Flex>
+              <Flex align='flex-end' flexDir='column' gap={2}>
+                <Skeleton width='120px' height='18px' />
+                <Skeleton width='80px' height='18px' />
+              </Flex>
+            </Flex>
+          ))}
+        </Flex>
+      ) : (
+        <Flex
+          flexDir='column'
+          width='100%'
+          maxHeight={`calc(100vh - var(--mobile-nav-offset) - ${bodyHeaderHeight})`}
+          overflowY='scroll'
+          px={4}
+        >
+          <InfiniteTable
```

Comment:
For simplicity, used the same component we already use everywhere, even thought its not needed but mainly to reuse `AssetCell` without the need to build a table by myself!